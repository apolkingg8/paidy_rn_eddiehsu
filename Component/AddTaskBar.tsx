import {Pressable, StyleSheet, Text, TextInput, View} from 'react-native';
import {useState} from "react";
import {observer} from "mobx-react";
import Task from "../Klass/Task";
import taskStore from "../store/taskStore";
import {percent} from "csx";
import styleStore from "../store/styleStore";

const AddTaskBar = () => {
    const [text, setText] = useState('')

    return (
        <View style={styles.wrap}>
            <TextInput
                style={styles.input}
                selectionColor={styleStore.palette.primary}
                placeholderTextColor={styleStore.palette.lightGray}
                placeholder={'Some simple note'}
                maxLength={96}
                onChangeText={(newText)=> {
                    setText(newText)
                }}
                value={text}
            />
            <Pressable
                style={styles.button}
                onPress={()=> {
                    if(text.length === 0) {
                        return
                    }

                    const newTask = new Task({
                        text: text,
                    })
                    taskStore.addTask(newTask)
                    setText('')
                }}
            >
                <Text>ADD</Text>
            </Pressable>
        </View>
    )
}

const styles = StyleSheet.create({
    wrap: {
        ...styleStore.centerRow,
        width: percent(100),
        height: 40,
        marginTop: 48,
        marginBottom: 24,
        paddingHorizontal: 24,
    },
    input: {
        flex: 1,
        height: percent(100),
        padding: 8,
        color: styleStore.palette.white,
        borderWidth: 1,
        borderColor: styleStore.palette.white,
    },
    button: {
        ...styleStore.centerRow,
        height: percent(100),
        marginLeft: 8,
        paddingHorizontal: 24,
        backgroundColor: styleStore.palette.primary,
    },
    buttonText: {
        color: styleStore.palette.white,
    },
})

export default observer(AddTaskBar)
