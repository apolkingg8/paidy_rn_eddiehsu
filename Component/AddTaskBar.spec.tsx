import {render} from "@testing-library/react-native";
import AddTaskBar from "./AddTaskBar";

jest.mock('expo-local-authentication')

describe('AddTaskBar', ()=> {
    test('render', ()=> {
        const component = render(<AddTaskBar/>).toJSON()
        expect(component).toMatchSnapshot()
    })
})
