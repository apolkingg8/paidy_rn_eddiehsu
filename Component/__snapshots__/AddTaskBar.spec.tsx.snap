// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTaskBar render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "center",
      "marginBottom": 24,
      "marginTop": 48,
      "paddingHorizontal": 24,
      "width": "100%",
    }
  }
>
  <TextInput
    maxLength={96}
    onChangeText={[Function]}
    placeholder="Some simple note"
    placeholderTextColor="#959595"
    selectionColor="#b0e55a"
    style={
      Object {
        "borderColor": "#f7f7f7",
        "borderWidth": 1,
        "color": "#f7f7f7",
        "flex": 1,
        "height": "100%",
        "padding": 8,
      }
    }
    value=""
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#b0e55a",
        "display": "flex",
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "center",
        "marginLeft": 8,
        "paddingHorizontal": 24,
      }
    }
  >
    <Text>
      ADD
    </Text>
  </View>
</View>
`;
